File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let answer = 1;
    const TYPE = new Set(clothes.map((v) => v[1]));
    [...TYPE].forEach((value) => {
        answer *= (clothes.filter((val) => val[1] === value).length + 1);
    });
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothesList):
    total = 1
    hash = {}

    for clothes in clothesList:
        for i in range(len(clothes) - 1):
            if clothes[-1] in hash:
                hash[clothes[-1]] += 1
            else:
                hash[clothes[-1]] = 2

    for clothes in hash:
        total *= hash[clothes]

    return total - 1